common/src/androidTest/java/com/microsoft/identity/common/AzureActiveDirectoryAuthorizationResultFactoryTest.java

```diff
@@ -0,0 +1,236 @@
+package com.microsoft.identity.common;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.test.runner.AndroidJUnit4;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationResult;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationResultFactory;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStatus;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+
+@RunWith(AndroidJUnit4.class)
+public class AzureActiveDirectoryAuthorizationResultFactoryTest {
+
+    private static final String REDIRECT_URI = "aad-clientid://packagename/";
+    private static final String AUTH_CODE_AND_STATE = "code=authorization_code&state=state";
+    private static final String AUTH_CODE = "authorization_code";
+    private static final String STATE = "state";
+    private static final String ERROR_MESSAGE = "access_denied";
+    private static final String ERROR_DESCRIPTION = "access denied error description";
+    private static final String CORRELATION_ID = "correlationId";
+    private static final String ERROR_CODES = "access_denied_error_code";
+
+    @Test
+    public void testBrowserCodeCancel() {
+        Intent intent = new Intent();
+        intent.putExtras(new Bundle());
+        AzureActiveDirectoryAuthorizationResultFactory factory = new AzureActiveDirectoryAuthorizationResultFactory();
+        AuthorizationResult result = factory.createAuthorizationResult(AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL, intent);
+        assertTrue(result instanceof AzureActiveDirectoryAuthorizationResult);
```

Comment:
See https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/133/files/34372525a1342e7fc0e5ff69a49a0e871423275a#r195851360 -- we can avoid these checks/casts

Reply:
Love it excellent suggestion @iambmelt  :ok_hand:. Made changes as required